addons/web_editor/static/src/components/media_dialog/search_media.js

```diff
@@ -1,21 +1,31 @@
 /** @odoo-module **/
 
-import { debounce } from '@web/core/utils/timing';
+import { useDebounced } from '@web/core/utils/timing';
 import { useAutofocus } from '@web/core/utils/hooks';
 
-import { Component, onWillUnmount, xml } from "@odoo/owl";
+import { Component, xml, useEffect, useState } from "@odoo/owl";
 
 export class SearchMedia extends Component {
     setup() {
         useAutofocus();
-        this.search = debounce((ev) => this.props.search(ev.target.value), 1000);
-        onWillUnmount(() => {
-            this.search.cancel();
+        this.debouncedSearch = useDebounced(this.props.search, 1000);
+
+        this.state = useState({
+            input: '',
         });
+
+        useEffect((input) => {
+            // Do not trigger a search on the initial render.
+            if (this.hasRendered) {
+                this.debouncedSearch(input);
+            } else {
+                this.hasRendered = true;
+            }
+        }, () => [this.state.input]);
     }
 }
 SearchMedia.template = xml`
 <div class="position-relative mw-lg-25 flex-grow-1 me-auto">
-    <input type="text" class="o_we_search o_input form-control" t-att-placeholder="props.searchPlaceholder.trim()" t-att-value="props.needle" t-on-input="search" t-ref="autofocus"/>
+    <input type="text" class="o_we_search o_input form-control" t-att-placeholder="props.searchPlaceholder.trim()" t-model="state.input" t-ref="autofocus"/>
```

Comment:
About the `t-model`, see review on previous PR: https://github.com/odoo/odoo/pull/104590#discussion_r1009181172